File: subroutines/procedure-function/c-sharp/calculate_area.cs
/*
Raspberry Pi Foundation
Developed to be used alongside Isaac Computer Science, part of the National Centre for Computing Education
Usage licensed under CC BY-SA 4

Note: This file is designed to be copied out and compiled on your machine.
In order for it to compile properly you need to ensure that the project name is the same as the "namespace" in this file.
To run this file you need to:
1. Copy the contents
2. Paste them into the C# IDE of your choice (Visual Studio, for example)
3. Change the namespace to match your project (if necessary)
4. Compile the program
5. Run the program
*/

using System;

namespace IsaacCodeSamples 
{
    class ProceduresFunctions
    {
        // The Main method is the entry point for all C# programs
        public static void Main()
        {
            float length = 12;
            float result = CalculateArea(length);
            Console.WriteLine(result);
        }


        // Calculates the area of a square with sides of a given length
        public static float CalculateArea(float length)
        {
            float area = length * length;
            return area;
        }


    }
}

File: subroutines/procedure-function/visual-basic/calculate_area.vb
' Raspberry Pi Foundation
' Developed as part of Ada Computer Science
' Usage licensed under CC BY-NC-SA 4.0
' Note: This file is designed to be copied out and compiled on your machine.
' To run this file you need to:
' 1. Copy the contents
' 2. Paste them into the Visual Basic IDE of your choice (Visual Studio, for example)
' 3. Compile the program
' 4. Run the program


Module Program

    ' The Main subroutine is the default entry point for all VB programs
    Sub Main()
        Dim length As Single = 12
        Dim result = CalculateArea(length)
        Console.WriteLine(result)
    End Sub


    ' Calculates the area of a square with sides of a given length
    Function CalculateArea(ByVal length As Single) As Single
        Dim area = length * length
        Return area
    End Function


End Module
